ldms/src/store/darshan/darshan_stream_store.c

```diff
@@ -0,0 +1,486 @@
+{
+	int rc = 0;
+	sos_schema_t schema;
+
+	/* Close the container if it already exists */
+	if (sos)
+		sos_container_close(sos, SOS_COMMIT_ASYNC);
+
+
+	/* Check if the container at path is already present */
+	sos = sos_container_open(path, SOS_PERM_RW|SOS_PERM_CREAT, container_mode);
+	if (!sos) {
+		return errno;
+	}
+
+	app_schema = sos_schema_by_name(sos, darshan_data_template.name);
+	if (!app_schema) {
+		rc = create_schema(sos, &app_schema);
+		if (rc)
+			return rc;
+	}
+	return rc;
+}
+
+static const char *usage(struct ldmsd_plugin *self)
+{
+	return	"config name=darshan_stream_store path=<path> port=<port_no> log=<path>\n"
+		"     path	The path to the root of the SOS container store (required).\n"
+		"     stream	The stream name to subscribe to (defaults to 'darshan Connector').\n"
+		"     mode	The container permission mode for create, (defaults to 0660).\n";
+}
+
+static int stream_recv_cb(ldmsd_stream_client_t c, void *ctxt,
+			 ldmsd_stream_type_t stream_type,
+			 const char *msg, size_t msg_len,
+			 json_entity_t entity);
+static int config(struct ldmsd_plugin *self, struct attr_value_list *kwl, struct attr_value_list *avl)
+{
+	char *value;
+	char *producer_name;
+	int rc;
+	value = av_value(avl, "mode");
+	if (value)
+		container_mode = strtol(value, NULL, 0);
+	if (!container_mode) {
+		msglog(LDMSD_LERROR,
+		       "%s: ignoring bogus container permission mode of %s, using 0660.\n",
+		       darshan_stream_store.name, value);
+	}
+
+	value = av_value(avl, "stream");
+	if (value)
+		stream = strdup(value);
+	else
+		stream = strdup("darshanConnector");
+	ldmsd_stream_subscribe(stream, stream_recv_cb, self);
+
+	value = av_value(avl, "path");
+	if (!value) {
+		msglog(LDMSD_LERROR,
+		       "%s: the path to the container (path=) must be specified.\n",
+		       darshan_stream_store.name);
+		return ENOENT;
+	}
+
+	if (root_path)
+		free(root_path);
+	root_path = strdup(value);
+	if (!root_path) {
+		msglog(LDMSD_LERROR,
+		       "%s: Error allocating %d bytes for the container path.\n",
+		       strlen(value) + 1);
+		return ENOMEM;
+	}
+
+	rc = reopen_container(root_path);
+	if (rc) {
+		msglog(LDMSD_LERROR, "%s: Error opening %s.\n",
+		       darshan_stream_store.name, root_path);
+		return ENOENT;
+	}
+	return 0;
+}
+
+static int get_json_value(json_entity_t e, char *name, int expected_type, json_entity_t *out)
+{
+	int v_type;
+	json_entity_t a = json_attr_find(e, name);
+	json_entity_t v;
+	if (!a) {
+		msglog(LDMSD_LERROR,
+		       "%s: The JSON entity is missing the '%s' attribute.\n",
+		       darshan_stream_store.name,
+		       name);
+		return EINVAL;
+	}
+	v = json_attr_value(a);
+	v_type = json_entity_type(v);
+	if (v_type != expected_type) {
+		msglog(LDMSD_LERROR,
+		       "%s: The '%s' JSON entity is the wrong type. "
+		       "Expected %d, received %d\n",
+		       darshan_stream_store.name,
+		       name, expected_type, v_type);
+		return EINVAL;
+	}
+	*out = v;
+	return 0;
+}
+
+
+// Metadata
```

Comment:
Is this supposed to be here? If it is a comment, please use /*

Reply:
@aazizo what is the status of this pull request? Have you made the requested changes?

Reply:
@tom95858 I made the changes but I asked you a question about the timestamp overlapping with the job_id in the below section. Can you please answer it so I can do more changes if necessary. I am confused what do you mean by the timestamp overlap the job _id field?
I am also learning how to do TADA testing so I can perform it to this code.